app/celery/process_pinpoint_receipts_tasks.py

```diff
@@ -0,0 +1,145 @@
+@statsd(namespace="tasks")
+def process_pinpoint_results(self, response):
+    try:
+        receipt = json.loads(response["Message"])
+        reference = receipt["messageId"]
+        status = receipt["messageStatus"]
+        provider_response = receipt["messageStatusDescription"]
+
+        notification_status = determine_pinpoint_status(status, provider_response)
+        if not notification_status:
+            current_app.logger.warning(f"unhandled provider response for reference {reference}, received '{provider_response}'")
+            notification_status = NOTIFICATION_TECHNICAL_FAILURE  # revert to tech failure by default
+
+        try:
+            notification = notifications_dao.dao_get_notification_by_reference(reference)
+        except NoResultFound:
+            try:
+                current_app.logger.warning(
+                    f"RETRY {self.request.retries}: notification not found for Pinpoint reference {reference} (update to {notification_status}). "
+                    f"Callback may have arrived before notification was persisted to the DB. Adding task to retry queue"
+                )
+                self.retry(queue=QueueNames.RETRY)
+            except self.MaxRetriesExceededError:
+                current_app.logger.warning(
+                    f"notification not found for Pinpoint reference: {reference} (update to {notification_status}). Giving up."
+                )
+            return
+        if notification.sent_by != PINPOINT_PROVIDER:
+            current_app.logger.exception(f"Pinpoint callback handled notification {notification.id} not sent by Pinpoint")
+            return
+
+        if notification.status != NOTIFICATION_SENT:
+            notifications_dao._duplicate_update_warning(notification, notification_status)
+            return
+
+        notifications_dao._update_notification_status(
+            notification=notification,
+            status=notification_status,
+            provider_response=provider_response,
+        )
+
+        if notification_status != NOTIFICATION_DELIVERED:
+            current_app.logger.info(
+                (
+                    f"Pinpoint delivery failed: notification id {notification.id} and reference {reference} has error found. "
+                    f"Provider response: {provider_response}"
+                )
+            )
+        else:
+            current_app.logger.info(
+                f"Pinpoint callback return status of {notification_status} for notification: {notification.id}"
+            )
+
+        statsd_client.incr(f"callback.sns.{notification_status}")  # TODO: do we want a Pinpoint metric here?
+
+        if notification.sent_at:
+            statsd_client.timing_with_dates(
+                "callback.sns.elapsed-time", datetime.utcnow(), notification.sent_at
+            )  # TODO: do we want a Pinpoint metric here?
+
+        _check_and_queue_callback_task(notification)
+
+        return True
```

Comment:
Why do we return true here, but simple return (no value) for others? This was also picked up by our scanner over here:
https://github.com/cds-snc/notification-api/security/code-scanning/804

Reply:
Yeah, this is a good question. It's what the other tasks do, and all I could think of was that the tests were returning `None` if they failed and this `True` would indicate a success. My googling wasn't able to confirm or dispute this theory however :/

Reply:
Hmm I wonder if this function was called elsewhere maybe and hence expected to return something, maybe the older CLI commands? 🤔

Reply:
ha! I was excluding our tests when looking to see how it's called. Our tests use the return value to determine if the function successfully processed the receipt (`True`) or if there was a problem (`None`).

I'll add a `return` at the end of the function to quiet down the scanner.

Reply:
I would expect the tests to be able to determine the end by verifying if the `_check_and_queue_callback_task` function was called but the code wanted it simpler I guess. 

Do you still want to remove that `True` here? 👀

Reply:
yeah... having code in the app just for test verification is ugly and also could lead to tests passing when it doesn't really work (like if there was a `return True` where there shouldn't be). I changed the tests to just checking that callback task as you suggested.